src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.cs

```diff
@@ -377,6 +384,99 @@ public void WriteValue(string value)
             this.FlushIfBufferThresholdReached();
         }
 
+        private void WriteStringValueInChunks(ReadOnlySpan<char> value)
+        {
+            int chunkSize = bufferWriter.FreeCapacity;
+
+            for (int i = 0; i < value.Length; i += chunkSize)
+            {
+                int remainingChars = Math.Min(chunkSize, value.Length - i);
+                ReadOnlySpan<char> chunk = value.Slice(i, remainingChars);
+
+                if (NeedsEscaping(chunk) != -1)
+                {
+                    WriteEscapedStringChunk(chunk);
+                }
+                else
+                {
+                    WriteStringChunk(chunk);
+                }
+
+                // Flush the buffer if needed
+                this.FlushIfBufferThresholdReached();
+            }
+        }
+
+        private int NeedsEscaping(ReadOnlySpan<char> value)
+        {
+            if (value.IsEmpty)
+            {
+                return -1;
+            }
+
+            // Iterate through each character in the span
+            for (int i = 0; i < value.Length; i++)
+            {
+                string encoded = this.encoder.Encode(value[i].ToString());
+                
+                // if encoded string is different from original string then we know special
+                // characters exist and need escaping.
+                if (!string.Equals(value[i].ToString(), encoded, StringComparison.Ordinal))
+                {
+                    return i;
+                }
+            }
+
+            return -1;
+        }
+
+        private void WriteEscapedStringChunk(ReadOnlySpan<char> sourceChunk)

```

Comment:
I think you should pass the index of the first character to index here so that the characters before that index can be copied directly without escaping.